Create pie chart

KuiMing opened the pull request:
1. Create Pie Chart. echart(x,~name, ~value,type='pie')
2. Data removable and calculable.
3. MagicType, restore and saveAsImage was added.

KuiMing committed:
tool tip test

KuiMing committed:
back

KuiMing committed:
Try it again

KuiMing committed:
tooltip failed

KuiMing committed:
Pie plot test

KuiMing committed:
for data_pie

KuiMing committed:
data_pie again

KuiMing committed:
delete data_pie

KuiMing committed:
found the bug and try it again

KuiMing committed:
forgot "pie"

KuiMing committed:
Pie and funnel

KuiMing committed:
put series into pie directly

KuiMing committed:
Test

KuiMing committed:
bug is series

KuiMing committed:
all the problems are about data structure

KuiMing committed:
add tooltip, legend and magicType to pie plot

KuiMing committed:
remove frame of Pie

KuiMing committed:
Revert "remove frame of Pie"

This reverts commit 7ee1b35fcfb8ffb0a0cd6a75e2901b5d1310cd05.

KuiMing committed:
try to remove frame of pie

KuiMing committed:
Revert "try to remove frame of pie"

This reverts commit a5425b605d1406e429444cf4d16f9ca604753c5a.

KuiMing committed:
Remove title
Add magictype function

KuiMing committed:
calculable control add or remove data

KuiMing committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	R/echart.R

KuiMing committed:
remove unnecessary calculable

yihui commented on R/example.R:
```diff
@@ -0,0 +1,43 @@
+pieEx1 = list(
```
This does not look like a good example in my eyes. Ideally we should not require users to prepare a huge list for `echart()`, otherwise that defeats the purpose of this package. Instead, we want something as simple as `echart(data, ~variable, type = 'pie')`. Do you have a better example?

KuiMing replied:
Sorry, it's my fault. It was just for testing on my own. I will give a simple example.

yihui commented on R/echart.R:
```diff
@@ -63,16 +63,57 @@ echart.data.frame = function(
     x = x, y = y
   ), TOJSON_ARGS = list(pretty = TRUE))
 
+  params$tooltip = list(
```
I'd prefer a separate function to do these configurations, e.g. something like `eChart() %>% eConfig(calculable = TRUE) %>% eToolbox(show = TRUE, feature = ...)`. It is fine to hard-code some options inside `echart()`, but we should make it possible for users to change these options at the same time.

KuiMing replied:
I will give a new function for options.

yihui commented:
Two more things:
1. Please provide some screenshots.
2. Eventually I will need you to clean up the commit history. If you are not very familiar with GIT, I have some instructions here: https://github.com/yihui/knitr/blob/master/CONTRIBUTING.md#pull-requests

Thanks!

KuiMing committed:
Improvement according to comment at 2015/5/14

1. give a real example code for pie chart.
2. give a pie.html
3. remove unnecessary code of data.R and echart.R
4. add eConfig and etoolbox in option.R

KuiMing commented:
Improvement according to comment at 2015/5/14
1. give a real example code for pie chart.
2. give a pie.html
3. remove unnecessary code of data.R and echart.R
4. add eConfig and etoolbox in option.R

KuiMing commented:
One more thing, I will rebase after you confirm these are ok.

KuiMing commented:
Hello Yihui,

Did you check my new code? Is there anything I need to modify?

KuiMing committed:
candlestick and modification of pie

1. create candlestick chart
2. modify pie chart

KuiMing committed:
update

version update
egauge
